storagenode/contact/chore.go

```diff
@@ -48,60 +58,111 @@ func (chore *Chore) Run(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
 	chore.log.Info("Storagenode contact chore starting up")
 
+	var group errgroup.Group
+
 	if !chore.service.initialized.Wait(ctx) {
 		return ctx.Err()
 	}
 
-	return chore.Loop.Run(ctx, func(ctx context.Context) error {
-		if err := chore.pingSatellites(ctx); err != nil {
-			chore.log.Error("pingSatellites failed", zap.Error(err))
-		}
-		return nil
-	})
+	initialBackOff := time.Second
+	interval := chore.interval
+	chore.mu.Lock()
+	for _, satellite := range chore.trust.GetSatellites(ctx) {
+		satellite := satellite
+
+		cycle := sync2.NewCycle(interval)
+		chore.cycles = append(chore.cycles, cycle)
+
+		cycle.Start(ctx, &group, func(ctx context.Context) error {
+			chore.log.Info("contact cycle: satellite " + satellite.String() + " - starting contact cycle")
```

Comment:
should it be at the debug level?

Reply:
Yeah, debug sounds better. Also no need to mention "contact cycle", because all the log statements will have "contact:chore" or similar always prepended to it anyways.